core/signer/signer.go

```diff
@@ -310,9 +351,22 @@ func (s *Signer) CreateEpoch(ctx context.Context, forceNewEpoch bool) error {
 	glog.V(2).Infof("CreateEpoch: SetLeaves:{Revision: %v, HighestFullyCompletedSeq: %v}", revision, seq)
 
 	// Put SignedMapHead in an append only log.
-	if err := s.sths.Write(ctx, s.logID, revision, setResp.GetMapRoot()); err != nil {
-		// TODO(gdbelvin): If the log doesn't do this, we need to generate an emergency alert.
-		return fmt.Errorf("sths.Write(%v, %v): %v", s.logID, revision, err)
+	smrJSON, err := json.Marshal(setResp.GetMapRoot())
+	if err != nil {
+		return err
+	}
+	idHash := sha256.Sum256(smrJSON)
+	// TODO(gbelvin): Add leaf at a specific index. trillian#423
+	// TODO(gdbelvin): If the log doesn't do this, we need to generate an emergency alert.
+	if _, err := s.tlog.QueueLeaf(ctx, &trillian.QueueLeafRequest{
+		LogId: s.logID,
+		Leaf: &trillian.LogLeaf{
+			LeafValue:        smrJSON,
+			LeafIdentityHash: idHash[:],
+		},
+	}); err != nil {
+		return fmt.Errorf("trillianLog.QueueLeaf(logID: %v, leaf: %v): %v",
+			s.logID, smrJSON, err)
```

Comment:
Good call.  Eventually we will replace this code with a proper Verifying Log Client that auto configures itself